### ISSUE TITLE
Juju failed to bootstrap: maas

### ISSUE THREAD

**Issue by merlin2002** [conjure-up/conjure-up#817]
```
2017-04-12 23:21:57,038 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: LXD version: 2.12, Juju version: 2.2-beta3-xenial-amd64
2017-04-12 23:22:10,929 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Path is local filesystem, copying /snap/conjure-up/189/spells/landscape to /home/heping/.cache/conjure-up/landscape
2017-04-12 23:22:10,933 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Pulling bundle for landscape-dense-maas from channel: stable
2017-04-12 23:23:44,829 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Performing bootstrap: conjure-up-maas-621 maas/192.168.232.140
2017-04-12 23:23:44,831 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: bootstrap cmd: juju bootstrap maas/192.168.232.140 conjure-up-maas-621 --config image-stream=daily --config enable-os-upgrade=false --default-model conjure-landscape-584 --to main-condor --bootstrap-series=xenial --credential conjure-up-maas-621 --debug
2017-04-12 23:24:37,460 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: handle bootstrap
2017-04-12 23:24:37,461 [ERROR] log.py:17 - conjure-up/_unspecified_spell: 23:23:44 INFO  juju.cmd supercommand.go:63 running juju [2.2-beta3 gc go1.6.2]

23:23:44 DEBUG juju.cmd supercommand.go:64   args: []string{"juju", "bootstrap", "maas/192.168.232.140", "conjure-up-maas-621", "--config", "image-stream=daily", "--config", "enable-os-upgrade=false", "--default-model", "conjure-landscape-584", "--to", "main-condor", "--bootstrap-series=xenial", "--credential", "conjure-up-maas-621", "--debug"}

23:23:44 INFO  cmd bootstrap.go:696 cloud "maas" not found, trying as a provider name

23:23:44 INFO  cmd bootstrap.go:721 interpreting "192.168.232.140" as the cloud endpoint

23:23:44 DEBUG juju.cmd.juju.commands bootstrap.go:786 authenticating with region "" and credential "conjure-up-maas-621" ()

23:23:44 DEBUG juju.cmd.juju.commands bootstrap.go:898 provider attrs: map[]

23:23:45 INFO  cmd authkeys.go:114 Adding contents of "/home/heping/.local/share/juju/ssh/juju_id_rsa.pub" to authorized-keys

23:23:45 INFO  cmd authkeys.go:114 Adding contents of "/home/heping/.ssh/id_rsa.pub" to authorized-keys

23:23:45 DEBUG juju.cmd.juju.commands bootstrap.go:954 preparing controller with config: map[apt-https-proxy: logforward-enabled:false apt-no-proxy:127.0.0.1,localhost,::1 apt-ftp-proxy: enable-os-upgrade:false agent-metadata-url: https-proxy: image-metadata-url: http-proxy: resource-tags: type:maas apt-http-proxy: apt-mirror: transmit-vendor-metrics:true provisioner-harvest-mode:destroyed disable-network-management:false automatically-retry-hooks:true ignore-machine-addresses:false authorized-keys:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC4DfDBORnNIb+Gw0mKLqRkCGey42+gAlg7Z09ZRr8q1VQt8Utzp8+vCxM0DTUMtA4iRI6TIfcl4o0pMbuHSmXeq2aINk/aG4WO3/nEdAp4zZ2UuimypUEovshiexeVbgXLB6vE/kgFtG8aF4pH6Qaoe+l6/dpHQp83ZKWzu4/cbgnpF8iCW7bsAtTHOkEQV1rPTSx0FPU+TdMgQhYyhdOMF/9Ps5XAdR1oS8a/bHlygHfP43mKWu/cEhS1peZUbUkFmabawZ9NntvdHko3B1k92lenJL2mVgrdh04a2CmJw0W2fNbmnCRYWAKtddQ8nTTmqT9vSbSFAE90djwHUCTd juju-client-key

ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDLkujTuMPXIVgHljn2qLrD3rM7m+UFWHrwjCHNQ7Hta1N33sSMqL2tP/ot4y4jThKirWmdqs0oLJlxszR9jwKhkLsgw0FgAMJgOsEaI/MiiVATFVA9xrntijOz5qPr2PXd56XZlPQdG8MAg45UxYNQ0l9O/Na8HSL19kMoUwQhcZJk2jQJ7D4I0UFph4bgtzR2CrI0ntTbiFjbwj6wMwib8Bn1OwSp6AT7tBtD2igvHtyUWSjC8e8algWZzDY+wDr5AIA89Y5whSFBp20aCU5dIRdAjxB1Ti9zjm6fN5sMf6N3nQH0W41PEWoJ8lbUQMpEjXhMhD8RKk4iANAHV5EZ heping@maas

 ssl-hostname-verification:true proxy-ssh:false name:controller uuid:62cbf82b-6764-47fc-8dae-4cb7519ec315 enable-os-refresh-update:true logging-config: no-proxy:127.0.0.1,localhost,::1 net-bond-reconfigure-delay:17 ftp-proxy: test-mode:false image-stream:daily default-series:xenial development:false agent-stream:released firewall-mode:instance]

23:23:45 DEBUG juju.provider.maas environprovider.go:56 opening model "controller".

23:23:45 INFO  cmd bootstrap.go:477 Creating Juju controller "conjure-up-maas-621" on maas

23:23:45 INFO  juju.cmd.juju.commands bootstrap.go:532 combined bootstrap constraints:

23:23:45 DEBUG juju.environs.bootstrap bootstrap.go:199 model "controller" supports service/machine networks: true

23:23:45 DEBUG juju.environs.bootstrap bootstrap.go:201 network management by juju enabled: true

23:23:45 INFO  cmd bootstrap.go:233 Loading image metadata

23:23:45 INFO  cmd bootstrap.go:296 Looking for packaged Juju agent version 2.2-beta3 for amd64

23:23:45 INFO  juju.environs.bootstrap tools.go:72 looking for bootstrap agent binaries: version=2.2-beta3

23:23:45 INFO  juju.environs.tools tools.go:101 finding agent binaries in stream "devel"

23:23:45 INFO  juju.environs.tools tools.go:103 reading agent binaries with major.minor version 2.2

23:23:45 INFO  juju.environs.tools tools.go:111 filtering agent binaries by version: 2.2-beta3

23:23:45 INFO  juju.environs.tools tools.go:114 filtering agent binaries by series: xenial

23:23:45 INFO  juju.environs.tools tools.go:117 filtering agent binaries by architecture: amd64

23:23:45 DEBUG juju.environs.tools urls.go:109 trying datasource "keystone catalog"

23:23:50 DEBUG juju.environs.simplestreams simplestreams.go:683 using default candidate for content id "com.ubuntu.juju:devel:tools" are {20161007 mirrors:1.0 content-download streams/v1/cpc-mirrors.sjson []}

23:24:13 INFO  juju.environs.bootstrap tools.go:74 found 0 packaged agent binaries

23:24:13 INFO  cmd bootstrap.go:315 No packaged binary found, preparing local Juju agent binary

23:24:13 DEBUG juju.environs.sync sync.go:302 Making agent binary tarball

23:24:13 DEBUG juju.environs.tools build.go:126 looking for: juju

23:24:13 DEBUG juju.environs.tools build.go:165 checking: /snap/conjure-up/189/bin/jujud

23:24:13 INFO  juju.environs.tools build.go:171 Found agent binary to upload (/snap/conjure-up/189/bin/jujud)

23:24:13 INFO  juju.environs.tools build.go:181 target: /tmp/juju-tools283799294/jujud

23:24:17 DEBUG juju.environs.tools build.go:254 forcing version to 2.2-beta3.1

23:24:17 DEBUG juju.environs.tools build.go:42 adding entry: &tar.Header{Name:"FORCE-VERSION", Mode:436, Uid:0, Gid:0, Size:11, ModTime:time.Time{sec:63627607457, nsec:170406209, loc:(*time.Location)(0x420f0c0)}, Typeflag:0x30, Linkname:"", Uname:"ubuntu", Gname:"ubuntu", Devmajor:0, Devminor:0, AccessTime:time.Time{sec:63627607457, nsec:170406209, loc:(*time.Location)(0x420f0c0)}, ChangeTime:time.Time{sec:63627607457, nsec:170406209, loc:(*time.Location)(0x420f0c0)}, Xattrs:map[string]string(nil)}

23:24:17 DEBUG juju.environs.tools build.go:42 adding entry: &tar.Header{Name:"jujud", Mode:509, Uid:0, Gid:0, Size:132946360, ModTime:time.Time{sec:63627607456, nsec:906405344, loc:(*time.Location)(0x420f0c0)}, Typeflag:0x30, Linkname:"", Uname:"ubuntu", Gname:"ubuntu", Devmajor:0, Devminor:0, AccessTime:time.Time{sec:63627607456, nsec:906405344, loc:(*time.Location)(0x420f0c0)}, ChangeTime:time.Time{sec:63627607456, nsec:906405344, loc:(*time.Location)(0x420f0c0)}, Xattrs:map[string]string(nil)}

23:24:35 INFO  juju.environs.sync sync.go:333 using agent binary 2.2-beta3-xenial-amd64 aliased to 2.2-beta3.1-xenial-amd64 (26083kB)

23:24:35 INFO  cmd bootstrap.go:357 Starting new instance for initial controller

Launching controller instance(s) on maas...

23:24:35 ERROR juju.cmd.juju.commands bootstrap.go:488 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=main-condor

23:24:35 DEBUG juju.cmd.juju.commands bootstrap.go:489 (error details: [{github.com/juju/juju/cmd/juju/commands/bootstrap.go:580: failed to bootstrap model} {github.com/juju/juju/provider/common/bootstrap.go:174: cannot start bootstrap instance} {github.com/juju/juju/provider/maas/environ.go:967: cannot run instances} {github.com/juju/juju/provider/maas/environ.go:1327: cannot run instance} {github.com/juju/juju/provider/maas/environ.go:768: } {github.com/juju/gomaasapi/controller.go:546: } {github.com/juju/gomaasapi/controller.go:743: } {github.com/juju/gomaasapi/controller.go:774: } {github.com/juju/gomaasapi/client.go:124: } {ServerError: 409 CONFLICT (No available machine matches constraints: mem=3584.0 name=main-condor)}])

23:24:35 DEBUG juju.cmd.juju.commands bootstrap.go:1061 cleaning up after failed bootstrap

23:24:35 INFO  juju.provider.common destroy.go:20 destroying model "controller"

23:24:35 INFO  juju.provider.common destroy.go:31 destroying instances

23:24:35 INFO  juju.provider.common destroy.go:51 destroying storage

23:24:35 INFO  cmd supercommand.go:465 command finished

2017-04-12 23:24:37,464 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Showing dialog for exception: Juju failed to bootstrap: maas
```

**Comment by battlemidget** [comment]
Your error

```
23:24:35 ERROR juju.cmd.juju.commands bootstrap.go:488 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=main-condor
```
Is related to juju not being able to find a suitable machine with those constraints. Can you verify that you are able to deploy nodes via MAAS webui and they come up successfully?

**Comment by merlin2002** [comment]
mem=3584.0 ：
machine memory less than 3584,Modify memory >=3584,it's ok
